### DISCUSSION TITLE
Application specific passwords

### DISCUSSION THREAD

**Discussion by giolekva** [ory/kratos#1940] [Q&A] [answered]
Hi all,
Does Kratos support associating multiple passwords with same user?
Such feature would enable legacy applications (with no OAuth2 support) to authenticate users using Kratos Basic Auth strategy while not leaking "master" password across all apps user is using.

If not, can someone outline what kind of changes will be necessary in Kratos to make this possible? And I'll take on implementing it.

Thanks.

**Comment by aeneasr** [comment, marked as answer]
That is currently not possible! Such a feature would definitely be quite difficult to implement though:

1. How do users know which password they are resetting when they recover their account?
2. If I change my password, which password do I change?
3. What password did I use to sign into app 1?
4. How do you reset passwords as an admin? Does it reset all of them? Or just one?

I think there are many more questions like that associated with this proposal. Do you have any examples where a system would support that to show how they solved this?

**Reply by vinckr** [reply]
Hey @giolekva ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
